README-dev.md

```diff
@@ -0,0 +1,244 @@
+This configuration allows changes to be made to both codebases.
+
+In the Alive-lsp directory:
+* Set the `alive.lsp.install.path` to the root of the Alive-lsp code.
+
+In the Alive directory:
+* Bring up VSCode.
+* Make changes to the Alive code.
+* Start the debugger (e.g. by pressing the `F5` key)
+  to generate a VSCode Extension Development Host window.
+* If necessary tell VSCode to open the Alive-lsp project in the Extension Development Host window.
+* Make changes to the Alive-lsp code.
+* Use the **Developer: Reload Window** command to restart the extension.
+* Exercise the Alive functionality supported by the server.
+* Log data from Alive-lsp will be in the `Output` view.
+
+#### Testing via REPL
+
+This method may help in cases where the REPL is crashing inexplicably.
+Consider this the configuration of last resort.
+
+In the Alive-lsp directory:
+* Do _not_ set the `alive.lsp.install.path` to the root of the Alive-lsp code.
+* Set the `alive.lsp.remote.host` to `"127.0.0.1"`.
+* Set the `alive.lsp.remote.port` to some number (e.g. `8006`).
+* Start the server in an SBCL REPL using the same port number:
+```
+        sbcl --eval "(asdf:load-system :alive-lsp)" \
+             --eval "(alive/server:start :port 8006)"
+```
+* Use the **Developer: Reload Window** command to restart the extension.
+* Exercise the server via the Alive extension to invoke the broken server code.
+  Some of this (e.g. acquiring defined packages)
+  may happen automatically at startup.
+* The REPL will crash and complain
+  `The current thread is not at the foreground`.
+* Execute `(sb-thread:release-foreground)`.
+  The REPL prompt may not be visible, just enter the command.
+* Use the Common Lisp
+  [interactive debugger](https://lispcookbook.github.io/cl-cookbook/debugging.html#the-interactive-debugger).
+
+The REPL can be run in a shell window from the Alive-lsp directory or in the
+`Terminal` view at the bottom of the VSCode workbench.
+Executing in a shell window may be marginally less confusing.
+
+With this configuration it is possible to add `(break)` calls in the code to cause a break in the REPL.
+Note that breaks in some areas will prevent the Alive extension from completing startup.
+For example, during startup Alive will request all known packages, ASDF systems, and threads.
+Breaking within the function `handle-list-pkgs` will halt processing of the packages request and
+the Alive extension startup will be delayed and (maybe) eventually fail.
+The **Locking Up the Alive Extension** section below addresses recovering from this sort of problem
+without resorting to `vim`.
+
+#### Debugging
+
+Debugging Lisp code, even with Alive, is different from most other languages.
+Breakpoints can not be marked in the editor window and stepped through in VSCode.
+The Common Lisp
+[interactive debugger](https://lispcookbook.github.io/cl-cookbook/debugging.html#the-interactive-debugger)
+is not directly available in VSCode.
```

Comment:
I'm probably not properly representing Alive's capabilities here.

Reply:
The debugger is available, but it's closer to what is available directly from SBCL than the SLDB debugger that Slime has. I'm still working on it, but the stack features aren't there, yet, and step debugging is flakey as hell in SBCL.

Reply:
I changed that to "still a work in progress". I can take the sentence out entirely if you would prefer.